standalone/pruntime/enclave/src/system/master_key.rs

```diff
@@ -1,33 +1,45 @@
-use crate::std::io::{Read, Write};
+use crate::std::convert::TryInto;
 use crate::std::path::PathBuf;
 use crate::std::prelude::v1::*;
 use crate::std::sgxfs::SgxFile;
 
-use phala_crypto::sr25519::{Persistence, Signing, SECRET_KEY_LENGTH, SIGNATURE_BYTES};
+use phala_crypto::sr25519::{Persistence, Signature, Signing, Sr25519SecretKey};
 use sp_core::sr25519;
 
+use serde::{Deserialize, Serialize};
+use serde_bytes;
+
 /// Master key filepath
 pub const MASTER_KEY_FILE: &str = "master_key.seal";
 
+#[derive(Debug, Deserialize, Serialize)]
```

Comment:
I think it's the time to think about the serialization methods. Now we have:

1. parity-scale-codec: used mostly for Substrate related logic (note that Ink! also uses it extensively and we may need to adopt it even more than now)
2. serde (cbor): used to seal the identity key and some basic config
3. serde (json): almost abandoned I guess
4. protobuf: prpc related logic

It looks like we'd better to only use the unavoidable dependencies (i.e. scale codec & protobuf).

@kvinwang What do you think?

Reply:
I agree to reduce dependencies. But SCALE and Protobuf both have  disadvantages.

For SCALE, the well-known issue is backward-compatibility. If we choose it, we might need to version the data with enum in advance.
For Protobuf, it feel like not that native, it's verbose decorations, less ability of expression than native rust types, and it's only for pRPC  in our project (not a direct dependency of enclaveapp).

So, it might better to choose Versioned SCALE codec?

Reply:
I think so. `enum MasterKeySeal { V1 { ... } }` looks good. Serde cbor also has the same versioning problem I think. So adding version is necessary anyway.